packages/devtools_app/lib/src/shared/table/table.dart

```diff
@@ -811,6 +806,25 @@ class _TableState<T> extends State<_Table<T>> with AutoDisposeMixin {
     scrollController = widget.tableController.verticalScrollController!;
 
     pinnedScrollController = ScrollController();
+  }
+
+  @override
+  void didUpdateWidget(covariant _Table<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (_data == oldWidget.tableController.tableData.value.data) return;
```

Comment:
we need to check that `widget.tableController.tableData == oldWidget.tableController.tableData` here. We aren't checking for equality of the data, but rather equality of the ValueNotifier.

Also this early return isn't exactly accurate. This only matters for the `widget.tableController.tableData` listener. But the tableData notifier could be unchanged in theory and `widget.selectionNotifier` could have been changed, which would mean we'd need to call `_initScrollOnSelectionListener(widget.selectionNotifier);` still. 

For simplicity, we can change this if check to check all three conditions:

```
if (widget.tableController.tableData != oldWidget.tableController.tableData || widget.selectionNotifier != oldWidget.selectionNotifier || widget.activeSearchMatchNotifier != oldWidget.activeSearchMatchNotifier) {
  cancelListeners();
  // init all three listeners
}
```